README.md

```diff
@@ -2,25 +2,284 @@
+
+-- Using the power of Monad we can choose which Dynamic to observe.
+which :: Dynamic Bool
+(which >>= if _ then d1 else d2) :: Dynamic Int
+```
+
+We can introduce new Dynamics using `newDynamic`:
+
+```purescript
+-- | Construct a new root Dynamic that can be changed from `Effect`-land.
+newDynamic :: forall m a. MonadEffect m => a -> m { dynamic :: Dynamic a, read :: Effect a, set :: a -> Effect Unit, modify :: (a -> a) -> Effect Unit }
+```
+
+#### [Event](https://pursuit.purescript.org/packages/purescript-specular/docs/Specular.FRP.Base#t:Event)
+
+`Event a` represents a source of occurences. Each occurence carries a value of type `a`.
+
+`Event` is a `Functor`.
+
+We can construct a trivial event `never :: forall a. Event a`, which never occurs.
+
+Events can be combined:
+
+```purescript
+-- | An Event that occurs when any of the events occur. If some of them occur simultaneously, the occurence value is that of the leftmost one.
+leftmost :: forall a. Array (Event a) -> Event a
+```
+
+Events can be transformed:
+
+```purescript
+-- | Retain only the occurences of the event for which the given predicate function returns `true`.
+filterEvent :: forall a. (a -> Boolean) -> Event a -> Event a
+
+-- | Map the given function over an Event, and retain only the occurences for which it returned a Just value.
+filterMapEvent :: forall a b. (a -> Maybe b) -> Event a -> Event b
+
+-- | Retain only the occurences of the Event which contain a Just value.
+filterJustEvent :: forall a. Event (Maybe a) -> Event a
+```
+
+We can observe `Event`s by being notified of their occurences.
+
+```purescript
+-- | Execute the given action for each occurence of the Event.
+subscribeEvent_ :: forall m a. MonadEffect m => MonadCleanup m => (a -> Effect Unit) -> Event a -> m a
+```
+
+#### [Callback](https://pursuit.purescript.org/packages/purescript-specular/docs/Specular.Callback#t:Callback)
+
+`Callback a` represents an action handler which consumes a value of type `a`. Think of it as `a -> Effect Unit`.
+
+```purescript
+-- | Trigger the action in Effect.
+triggerCallback :: forall a. Callback a -> a -> Effect Unit
+```
+
+#### [Ref](https://pursuit.purescript.org/packages/purescript-specular/docs/Specular.Ref#t:Ref)
+
+`Ref a` represents a read-write reference to a mutable observable variable.
+
+We can think of a `Ref` as of `Effect.Ref`, but with additional functions:
+- the ability to notify subscribers about changes to the value,
+- the ability to focus using a lens.
+
+`Ref a` consists of:
+- `Ref.value :: Ref a -> Dynamic a` to observe the value
+- `Ref.modify :: Ref a -> Callback (a -> a)` to modify the value using a function
+
+As a shortcut we have `Ref.set :: Ref a -> Callback a` to replace the value completely.
+
+Creating a Ref:
+
+```purescript
+newRef :: forall m a. MonadEffect m => a -> m (Ref a)
+```
+
+`Ref` is not a `Functor`, because it's read-write. It's `Invariant`, that is, it can be mapped over using a bijection.
```

Comment:
This part is dense.

Reply:
Worth to mention `focusRef` & `pureFocusRef`. I found them extremely useful in my pet-project.
Also, add mention of `constRef` (equivalent of `pure` for dynamics)